src/NHibernate/Dialect/MsSqlCeDialect.cs

```diff
@@ -187,10 +188,15 @@ protected virtual void RegisterFunctions()
 			RegisterFunction("lower", new StandardSQLFunction("lower"));
 
 			RegisterFunction("trim", new AnsiTrimEmulationFunction());
+			RegisterFunction("iif", new SQLFunctionTemplate(null, "case when ?1 then ?2 else ?3 end"));
 
 			RegisterFunction("concat", new VarArgsSQLFunction(NHibernateUtil.String, "(", "+", ")"));
+			RegisterFunction("mod", new SQLFunctionTemplate(NHibernateUtil.Int32, "((?1) % (?2))"));
 
 			RegisterFunction("round", new StandardSQLFunction("round"));
+
+			RegisterFunction("bit_length", new SQLFunctionTemplate(NHibernateUtil.Int32, "datalength(?1) * 8"));
+			RegisterFunction("extract", new SQLFunctionTemplate(NHibernateUtil.Int32, "datepart(?1, ?3)"));
```

Comment:
All these added function were already tested (and previously failing) in `HQLFunctions` fixture.